src/Providers/ProviderRow.tsx

```diff
@@ -0,0 +1,122 @@
+import React, { JSXElementConstructor } from 'react';
+import { Link } from 'react-router-dom';
+import { RowProps } from 'src/components/TableView';
+import { CONDITIONS, PROVIDERS, useTranslation } from 'src/internal/i18n';
+import {
+  ACTIONS,
+  HOST_COUNT,
+  NAME,
+  NAMESPACE,
+  NETWORK_COUNT,
+  READY,
+  STORAGE_COUNT,
+  TYPE,
+  URL,
+} from 'src/utils/constants';
+
+import { PATH_PREFIX } from '@app/common/constants';
+import { StatusIcon } from '@migtools/lib-ui';
+import { ResourceLink } from '@openshift-console/dynamic-plugin-sdk';
+import { Button, Popover } from '@patternfly/react-core';
+import { DatabaseIcon, NetworkIcon, OutlinedHddIcon } from '@patternfly/react-icons';
+import { Td, Tr } from '@patternfly/react-table';
+
+import { MergedProvider } from './data';
+import { ProviderActions } from './providerActions';
+
+interface CellProps {
+  value: string;
+  entity: MergedProvider;
+  t?: (k: string) => string;
+}
+const StatusCell = ({ value, entity: { conditions }, t }: CellProps) => {
+  const existingConditions = Object.values(conditions).filter(Boolean);
+  const toState = (value) => {
+    switch (value) {
+      case 'True':
+        return 'Ok';
+      case 'False':
+        return 'Error';
+      default:
+        return 'Unknown';
+    }
+  };
+  const label = CONDITIONS?.[value]?.(t);
```

Comment:
So is the initial `?.` really required in these cases?
```js
  const label = CONDITIONS[value]?.(t);
```

Reply:
Looking at the whole flow, you are right - at this point `value` should be a valid enum value.
However, looking only at this component it does add some extra safety - `value` is just a string and can be an invalid.
Instead of removing I would add here some meaningful fallback i.e. `Unknown`. What do you think?

Reply:
Ok it's two separate things now.

1. In this code `CONDITIONS?.[value]` and `CONDITION[value]` are equivalent since `CONDITIONS` is always defined and a constant.

2. `value` could _potentially_ be a value where `CONDITIONS[value]` is undefined.  That requires 2 things right?  The first thing is not to call `undefined` as a functions (covered with the `?.(t)` part).  The second thing is what should the value of `label` be?  So a meaningful fallback makes sense like `CONDITIONS[value]?.(t) ?? 'Unknown'`.